Fix travis

jovial opened the pull request:
Needed to specify pyflakes version early for:
```
    return self._retry_serial(self._styleguide.check_files, paths=paths)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/flake8/engine.py", line 172, in _retry_serial
    return func(*args, **kwargs)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pep8.py", line 1670, in check_files
    self.input_dir(path)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pep8.py", line 1706, in input_dir
    runner(os.path.join(root, filename))
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/flake8/engine.py", line 126, in input_file
    return fchecker.check_all(expected=expected, line_offset=line_offset)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pep8.py", line 1412, in check_all
    self.check_ast()
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pep8.py", line 1358, in check_ast
    checker = cls(tree, self.filename)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/flake8/_pyflakes.py", line 61, in __init__
    super(FlakesChecker, self).__init__(tree, filename,
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 294, in __init__
    self.handleChildren(tree)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 547, in handleChildren
    self.handleNode(node, tree)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 589, in handleNode
    handler(node)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 547, in handleChildren
    self.handleNode(node, tree)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 589, in handleNode
    handler(node)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 547, in handleChildren
    self.handleNode(node, tree)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 589, in handleNode
    handler(node)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 547, in handleChildren
    self.handleNode(node, tree)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 588, in handleNode
    handler = self.getNodeHandler(node.__class__)
  File "/home/will/code/stackhpc-inspector-plugins/.tox/pep8/lib/python3.8/site-packages/pyflakes/checker.py", line 458, in getNodeHandler
    self._nodeHandlers[node_class] = handler = getattr(self, nodeType)
AttributeError: 'FlakesChecker' object has no attribute 'CONSTANT'

```
See: https://github.com/PyCQA/pyflakes/issues/367

needed ironic client for:
```
ImportError: Failed to import test module: stackhpc_inspector_plugins.tests.unit.test_plugins_system_name_physnet
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/will/code/stackhpc-inspector-plugins/stackhpc_inspector_plugins/tests/unit/test_plugins_system_name_physnet.py", line 22, in <module>
    from stackhpc_inspector_plugins.plugins import system_name_physnet
  File "/home/will/code/stackhpc-inspector-plugins/stackhpc_inspector_plugins/plugins/system_name_physnet.py", line 20, in <module>
    from stackhpc_inspector_plugins.plugins import base_physnet
  File "/home/will/code/stackhpc-inspector-plugins/stackhpc_inspector_plugins/plugins/base_physnet.py", line 21, in <module>
    from ironicclient import exc as client_exc
ModuleNotFoundError: No module named 'ironicclient'

```

markgoddard committed:
Use py3 for tox by default, use train UC for python <= 3.5

Upstream OpenStack has dropped support for python 2, and even python
3.5. This means that master upper constraints no longer install on these
versions.

jovial committed:
Test with python 3.8

markgoddard commented:
Why py3.8? 3.8 isn't a supported runtime for OpenStack until Victoria. Train and Ussuri support 3.6-3.7.

jovial commented:
OK, add a job for 3.7 and 3.6 then?

markgoddard commented:
> OK, add a job for 3.7 and 3.6 then?

+1

jovial committed:
Test with python3.6 and python3.7

3.8 isn't a supported runtime for OpenStack until Victoria. Train and Ussuri support 3.6-3.7.

jovial committed:
Workaround for python3.8

jovial committed:
Whitespace fix
